includes/woopay/class-woopay-session.php

```diff
@@ -123,27 +123,54 @@ private static function get_user_id_from_cart_token() {
 		}
 
 		$cart_token = wc_clean( wp_unslash( $_SERVER['HTTP_CART_TOKEN'] ) );
+		$user_id    = self::get_payload_from_cart_token( $cart_token )->user_id;
 
-		if ( $cart_token && JsonWebToken::validate( $cart_token, '@' . wp_salt() ) ) {
-			$payload = JsonWebToken::get_parts( $cart_token )->payload;
+		if ( null === $user_id ) {
+			return null;
+		}
 
-			if ( empty( $payload ) ) {
-				return null;
-			}
+		$session_handler = new SessionHandler();
+		$session_data    = $session_handler->get_session( $user_id );
+		$customer        = maybe_unserialize( $session_data['customer'] );
```

Comment:
@alefesouza Have we considered the feasibility of keeping the verified user ID as an extra session data for the current session instead of introducing a different cart token?

I'm not sure if this is possible, or any better than the current approach, but I guess we could introduce a new key into the WooPay session handler similar to what you've done in `setup_session` in the tests, and then retrieve it here as the user ID in case it exists.

Alternatively, have we considered replacing the current session's user ID instead of introducing a new cart token?

Just dumping some thoughts before I tested anything thoughroughlly.

cc @asumaran as you might have insights into these questions as well.

Reply:
> the feasibility of keeping the verified user ID as an extra session data for the current session instead of introducing a different cart token?

You mean using [this line](https://github.com/Automattic/woocommerce-payments/blob/e45fc8418c3dadfa67e9c780419e25c0497f96cf/includes/class-wc-payments.php#L1513) and [this line](https://github.com/Automattic/woocommerce-payments/blob/add/allow-woopay-verified-email-requests/includes/woopay/class-woopay-session.php#L142) without the token verification? I'd prefer using the token to prevent someone from changing the session data (using some plugin maybe?) and then be logged in with the desired customer id.

Reply:
> the feasibility of keeping the verified user ID as an extra session data for the current session instead of introducing a different cart token?

@ricardo This PR was updated to use a verified email address header from WooPay on https://github.com/Automattic/woocommerce-payments/commit/ce78575f582990cde6d788be348d2834ce29d920, similar to this approach.